### PR TITLE
Change name from admin page.

### PR DESCRIPTION
**Pull request by nbianca**
Adds a button to change the name from admin page.

Fixes [this](https://meta.discourse.org/t/change-a-users-name-from-the-admin-page/59425).

### CODE REVIEW

**Comment by SamSaffron** [comment]
there is a slight consistency issue here. 

- Change username vs. Edit Name. We should just use one word here, not too, it should probably simply be "Edit Username" 

- Change username appears to work differently, this edits in place, change username edits on its own page. I would recommend changing "edit username" to be done the same way you did it for this field.

**Comment by nbianca** [comment]
@SamSaffron The "change" vs "edit" issue is found quite often. There are many fields using "change" (e.g. Change Username, Change About Me, Change Email, etc.) and about the same number of fields using "edit" (e.g. Edit Title, Edit Badges, etc.). I will change the "Change Username" to "Edit Username".

**Comment by YesThatAllen** [comment]
@SamSaffron which is a better match here... `Change Username` or `Edit Title` down below?

screenshot_4498

**Comment by nbianca** [comment]
This is how it looks now:

image

**Comment by SamSaffron** [comment]
cool, is there a leftover template and route for the previous way of editing username?

**Comment by nbianca** [comment]
That is still being used in user's preferences. Should that one be changed as well? Because if it is going to, there are many more fields that should be changed (email, user card badge, etc.)

File: app/assets/javascripts/admin/controllers/admin-user-index.js.es6 [hunk 1]
```diff
@@ -54,6 +56,44 @@ export default Ember.Controller.extend(CanCheckEmails, {
...
+    saveUsername() {
+      const self = this;
+
+      return ajax(`/users/${this.get('model.username').toLowerCase()}/preferences/username`, {
+        data: { new_username: this.get('userUsernameValue') },
+        type: 'PUT'
+      }).catch(function(e) {
+        bootbox.alert(I18n.t("generic_error_with_reason", {error: "http: " + e.status + " - " + e.body}));
+      }).finally(function() {
+        self.set('model.username', self.get('userUsernameValue'));
```
Comment (SamSaffron) [inline app/assets/javascripts/admin/controllers/admin-user-index.js.es6]: this is not really correct, if it fails to save finally will still be called and username will be changed incorrectly 

instead the pattern you should follow. 

1. amend model username
2. issue the PUT request
3. if there is an error, revert model username
4. finally toggle the edit

**Comment by SamSaffron** [comment]
@coding-horror  copy wise, do you prefer "Edit name/username" or "Change name/username" ?

**Comment by nbianca** [comment]
Made the changes after review. Would you like me to squash the commits?

**Comment by SamSaffron** [comment]
Sure, go for it, let's wait on @coding-horror to confirm what copy he would prefer here.

**Comment by coding-horror** [comment]
I actually feel _no_ copy is better here, just go with the pencil glyph on both, less to translate, less words on the screen for everyone to process in their mind grapes..

**Comment by SamSaffron** [comment]
@coding-horror what about "Edit Title" ... just a pencil there as well ?

**Comment by coding-horror** [comment]
Yeah just edit pencil button, much easier.

**Comment by nbianca** [comment]
After you click `Edit`, the input and a button labeled `Save {x}` appears. Should that be a glyph as well or should I keep the text in there?

**Comment by nbianca** [comment]
Done. The save button now simply says `Save`, not `Save {x}`.

File: app/assets/javascripts/admin/controllers/admin-user-index.js.es6 [hunk 2]
```diff
@@ -55,21 +57,65 @@ export default Ember.Controller.extend(CanCheckEmails, {
...
+
+    saveUsername() {
+      const self = this;
+      const old_username = self.get('model.username');
+
+      self.set('model.username', self.get('userUsernameValue'));
+      return ajax(`/users/${old_username.toLowerCase()}/preferences/username`, {
+        data: { new_username: this.get('userUsernameValue') },
+        type: 'PUT'
+      }).catch(function(e) {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-user-index.js.es6]: We prefer to use [Arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) in our code base.

File: app/assets/javascripts/admin/controllers/admin-user-index.js.es6 [hunk 3]
```diff
@@ -55,21 +57,65 @@ export default Ember.Controller.extend(CanCheckEmails, {
...
+    },
+
+    toggleNameEdit() {
+        this.set('userNameValue', this.get('model.name'));
+        this.toggleProperty('editingName');
+    },
+
+    saveName() {
+      const self = this;
+      const old_name = self.get('model.name');
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-user-index.js.es6]: Style: We use camelCase for our variable names for JS files.

**Comment by tgxworld** [comment]
Merged in https://github.com/discourse/discourse/commit/9065df76ade33f763ef9b7dc5d225f874433d4f4

**Comment by tgxworld** [comment]
Aside from the minor comments, the PR looks good. Thanks @nbianca  :+1:

**Pull request closed without merging**